### PR TITLE
137 create storageservice interface and implement gcsstorageservice

### PR DESCRIPTION
**Pull request by nguzinski**
issue #137 

removed all hard coding of GCS and created an interface to allow for faster changes in the future.

changed to allow for faster swapping in the future, or general modification, further it follows best practice

created a storage_service.py file abstract class interface, to allow for inheritance of methods, Created a gcs_storage-service.py as a concrete implementation of the Interface, further I removed the hard-coded GCS from airsim, and replaced it with my concrete class. I may need to abstract slightly differently but for the current functionality of just uploading, this should be more than sufficent.

### CODE REVIEW

**Comment by nguzinski** [comment]
dont push this just yet, I came up with an idea for abstracting a little more

**Review by Damerson1 — COMMENTED** [review]
This looks fine to me. I would say just define uploadtoservice in the interface.

File: backend/PythonClient/multirotor/airsim_application.py [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
+from PythonClient.multirotor.monitor.gsc_storage_service import GCSStorageService
 
 class AirSimApplication:
     # Parent class for all airsim client side mission and monitors
     def __init__(self):
-        self.bucket_name = 'droneworld'  # The bucket created in GCS
-        self.storage_client = storage.Client.from_service_account_json('key.json')  # Initializes the GCS client
-        self.bucket = self.storage_client.bucket(self.bucket_name)  # Points to the GCS bucket
+        #implementation of the GCS service
+        self.gcs = GCSStorageService('droneworld')
```
Comment (mohamdlog) [inline backend/PythonClient/multirotor/airsim_application.py:15]: This should be self.storage_service = GCSStorageService('droneworld')

**Review by mohamdlog — CHANGES_REQUESTED** [review]
Focus less on concrete variable names (e.g., use storage_service instead of gcs) to follow DIP and make the code flexible. Keep the __init__.py files, as they define package structure and are used throughout the project.

**Comment by mohamdlog** [comment]
> dont push this just yet, I came up with an idea for abstracting a little more

Is there an update to this?

File: backend/PythonClient/multirotor/airsim_application.py (hunk 1 above)
Comment (nguzinski) [reply backend/PythonClient/multirotor/airsim_application.py:15]: changed

**Comment by nguzinski** [comment]
I looked into what I was thinking, but it is not something that won't be changed when we have to connect the interface to some of the other "upload' / "download" functions

**Review by mohamdlog — APPROVED** [review]
lgtm!

**Pull request merged**